src/main/java/org/truffleruby/core/encoding/EncodingNodes.java

```diff
@@ -490,26 +486,17 @@ public DynamicObject encodingGetObjectEncodingNil(DynamicObject object) {
     public static abstract class EncodingReplicateNode extends PrimitiveArrayArgumentsNode {
 
         @Specialization(guards = "isRubyString(nameObject)")
-        public DynamicObject encodingReplicate(VirtualFrame frame, DynamicObject self, DynamicObject nameObject,
-                @Cached("new()") SnippetNode snippetNode) {
+        public DynamicObject encodingReplicate(DynamicObject self, DynamicObject nameObject) {
             final String name = StringOperations.getString(nameObject);
             final Encoding encoding = EncodingOperations.getEncoding(self);
 
             final DynamicObject newEncoding = replicate(name, encoding);
             if (newEncoding == null) {
                 throw new RaiseException(coreExceptions().argumentErrorEncodingAlreadyRegistered(name, this));
             }
-
-            final Entry entry = getEntry(name);
-            snippetNode.execute(frame, "Encoding::EncodingMap[enc.name.upcase.to_sym] = [nil, index]", "enc", newEncoding, "index", entry.getIndex());
             return newEncoding;
         }
 
-        @TruffleBoundary
-        private Entry getEntry(final String name) {
```

Comment:
How are you handling the j.l.String case in Ruby?

Reply:
It's not needed, I just reuse another primitive to get the index based on `self.name`.

Reply:
Oh, but this is the index of the new encoding, I need to check that.